src/argilla/client/feedback/training/schemas.py

```diff
@@ -982,6 +1015,92 @@ def _prepare_for_training_with_trl(
         return ds
 
 
+class TrainingTaskForQuestionAnsweringFormat(BaseModel):
+    """
+    Union[Tuple[str, str, str], List[Tuple[str, str, str]]]
+    """
+
+    format: Union[Tuple[str, str, str], List[Tuple[str, str, str]]]
+
+
+class TrainingTaskForQuestionAnswering(BaseModel, TrainingData):
+    _formatting_func_return_types = TrainingTaskForQuestionAnsweringFormat
+    formatting_func: Optional[Callable[[Dict[str, Any]], Union[None, str, Iterator[str]]]] = None
+    question: Optional[TextField] = None
+    context: Optional[TextField] = None
+    answer: Optional[TextQuestion] = None
+
+    def _format_data(self, dataset: "FeedbackDataset") -> List[Dict[str, str]]:
+        if self.formatting_func is not None:
+            output = set()
+            for sample in dataset.format_as("datasets"):
+                question_context_answer = self.formatting_func(sample)
+                if question_context_answer is None:
+                    continue
+
+                self._test_output_formatting_func(question_context_answer)
+
+                if isinstance(question_context_answer, tuple):
+                    question_context_answer = {question_context_answer}
+
+                output |= set(question_context_answer)
```

Comment:
Q: Does this support `formatting_func` as a generator?

Reply:
In the example not, but should be updated, I did not think about updating it correctly.